testing/e2e/manifests/clientServer.go

```diff
@@ -0,0 +1,145 @@
+package manifests
+
+import (
+	_ "embed"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:embed embedded/client.go
+var clientContents string
+
+//go:embed embedded/server.go
+var serverContents string
+
+type testingResources struct {
+	Client  *appsv1.Deployment
+	Server  *appsv1.Deployment
+	Service *corev1.Service
+	Ingress *netv1.Ingress
+}
+
+func (t testingResources) Objects() []client.Object {
+	ret := []client.Object{
+		t.Client,
+		t.Server,
+		t.Service,
+		t.Ingress,
+	}
+
+	for _, obj := range ret {
+		setGroupKindVersion(obj)
+	}
+
+	return ret
+}
+
+func ClientAndServer(namespace, name, zoneName, nameserver, keyvaultURI string) testingResources {
+	host := strings.ToLower(namespace) + "." + zoneName
+
+	clientDeployment := newGoDeployment(clientContents, namespace, name+"-client")
+	clientDeployment.Spec.Template.Annotations["openservicemesh.io/sidecar-injection"] = "disabled"
+	clientDeployment.Spec.Template.Spec.Containers[0].Env = []corev1.EnvVar{{
+		Name:  "URL",
+		Value: "https://" + host,
+	},
+		{
+			Name:  "NAMESERVER",
+			Value: nameserver,
+		},
+		{
+			Name:      "POD_IP",
+			ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"}},
+		},
+	}
+	clientDeployment.Spec.Template.Spec.Containers[0].ReadinessProbe = &corev1.Probe{
+		FailureThreshold:    1,
+		InitialDelaySeconds: 1,
+		PeriodSeconds:       1,
+		SuccessThreshold:    1,
+		TimeoutSeconds:      5,
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path:   "/",
+				Port:   intstr.FromInt(8080),
+				Scheme: corev1.URISchemeHTTP,
+			},
+		},
+	}
+
+	serverName := name + "-server"
+	serverDeployment := newGoDeployment(serverContents, namespace, serverName)
+	serviceName := name + "service"
+	ingressName := name + "-ingress"
+
+	service :=
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
```

Comment:
you can move these service/ingress/etc. to the common file for reuse in the future

Reply:
Not sure how useful that is. Ingresses/services should pretty much always be bundled with the client server for our tests since that's how we actually test an ingress. I don't think we'll ever need a standalone one but if we did it would make sense to move to common. Any particular use you have in mind?

Reply:
Not a standalone service/ingress, this would just make it easier for reuse if we need it later on, and clean up these definitions, like you have the deployment creation function